### PR TITLE
Refactor code: Restructure command parse logic

### PR DESCRIPTION
**Pull request by YFshadaow**
Refactor code:
 - Let Parser only parse raw commands from string
 - Individual command parsing are now done in their own classes
 - Argument parsing restructured

### CODE REVIEW

**Comment by NeoMinWei** [comment]
LGTM, just need fix the CLI and update text-ui-test file

**Comment by wwweert123** [comment]
LGTM

**Review by ryan1604 — APPROVED** [review]
LGTM

**Review by wwweert123 — APPROVED** [review]
LGTM

**Comment by hshiah** [comment]
LGTM

**Review by NeoMinWei — APPROVED** [review]
LGTM

**Review by NeoMinWei — DISMISSED** [review]
Change entry to Cashflow, can see my PR for reference
Line 68 in Parser is deeply nested

File: src/main/java/seedu/financialplanner/commands/EntryCommand.java
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/EntryCommand.java:1]: thanks for your help on the exceptions

**Review by NeoMinWei — COMMENTED** [review]
LGTM

**Pull request merged**